src/libYARP_OS/src/BottleImpl.cpp

```diff
@@ -817,21 +817,6 @@ ConstString StoreDouble::toString() const
     } else {
```

Comment:
Whoa, not related to this fix but I did not know that YARP had something in place to deal with problems like https://github.com/robotology/idyntree/issues/288 .

Reply:
And apparently is a part that needs to be modified to compile on Android: https://alecive.github.io/blog/2015/08/31/YARP-Cross-Compilation/ .